### ISSUE TITLE
Implement Roon as a source

### ISSUE THREAD

**Issue by FoxxMD** [FoxxMD/multi-scrobbler#83]
https://roon.app/en/

https://github.com/RoonLabs/node-roon-api

**Comment by theikid** [comment]
Hey @FoxxMD do you know if this is possible? I'm scribbling to last.fm through Roon directly but would love to switch to listenbrainz instead. :)

**Comment by FoxxMD** [comment]
It seems possible but I'll need to get a trial account and (hopefully) have something implemented before it is over as I don't use Roon personally. I'll have more time in January to look into this.

**Comment by FoxxMD** [comment]
I have sent an email to roon support inquiring about an open-source program so I can develop/test

**Comment by basepi** [comment]
Did you ever hear back? I also found this thread you opened: https://community.roonlabs.com/t/id-like-to-develop-a-scrobbler-application-for-roon/278031

I'd be happy to sponsor the development of this feature by buying you a few months of a paid account, if you have time/interest. If not, maybe I'll try to find time to do it myself.

**Comment by FoxxMD** [comment]
Hi @basepi and thanks for the interest!

I had one interaction with roon support that pointed me towards their partner application but after submitting I never heard back from them.

<img width="1480" height="814" alt="Image" src="https://github.com/user-attachments/assets/c45fc52f-662d-4690-a888-fa5bd7ccea7a" />

While I appreciate the financial support offer I would definitely prefer to have you take a crack at it, for a few reasons:

* My bandwidth is limited and I already have [other tasks in the pipeline](https://github.com/users/FoxxMD/projects/4/views/2) that I want to get to first. [scrobble migrations](https://github.com/FoxxMD/multi-scrobbler/pull/390) will be a big task and I don't foresee having time for roon for at least a couple months
* On-going maintenance of the roon integration would be better owned by someone who 1) will continue to pay for a subscription and 2) will dogfood the integration

**Comment by basepi** [comment]
Understandable. I'll see if I can carve out some time.

**Comment by MrPowerMac** [comment]
I don’t have any development experience to speak of but I am an avid Roon user with a lifetime subscription and would be more than happy to help test out any work done on this. Being able to scrobble Roon directly to ListenBrainz without using last.fm as an intermediary would be great!

**Comment by FoxxMD** [comment]
@basepi also, thanks for bringing up that roon support thread. I didn't realize I had a reply from someone there. I don't know enough about rooExtend or the extensions developed by that user. Are extensions easy to install for roon users? If they could implement an extension to emulate [listenbrainz scrobbling with a custom URL](https://foxxmd.github.io/multi-scrobbler/configuration/sources/listenbrainz-endpoint/) it would solve this issue. That would definitely be easier than trying to implement a bespoke roon Source for MS, and keep the functionality dogfood-ed in the community as well as open up functionality to the official LZ instance.

I replied to DrCWO in that thread, hopefully they have some insight.

**Comment by MrPowerMac** [comment]
Extensions are fairly easy to set up, usually just running as a separate application that Roon discovers over the network. rooExtend is an extension manager that the author uses to distribute and manage his own extensions, which runs on a dedicated Raspberry Pi flashed with [one of his images](https://drive.google.com/drive/folders/1c72vVpuuVlIWJpB8flVAZlHXmW2KNd1k).
There is also [Roon Extension Manager](https://github.com/TheAppgineer/roon-extension-manager/wiki), which runs as a service that can be installed on a wider variety of Linux systems (I have it running in a dedicated DietPi VM).

Once extensions are running and Roon discovers them, they appear in a tab of the Roon settings:
<img width="1381" height="635" alt="Image" src="https://github.com/user-attachments/assets/79a7ae26-3cb8-4b47-8255-98bbb6de1457" />

Here is an example of what an extension settings menu looks like:
<img width="427" height="690" alt="Image" src="https://github.com/user-attachments/assets/99d4cdce-21a2-45e2-9fae-8c1a212ec8c0" />

**Comment by basepi** [comment]
Yeah, I was planning to do some work with an LLM and [pyroon](https://github.com/pavoni/pyroon) as a launching point -- it seems like it exposes now playing, including seek point, in a way that should be easy to use to match the plex source capabilities. Might even make sense to include configuration to whitelist/blacklist roon sources, that kind of thing.

And yes, it seems like it will be added via an extension in roon itself.

**Comment by basepi** [comment]
I've had decent success getting a multi-scrobbler source working for normal roon sources. Unfortunately, this is completely useless to me unless it also supports scrobbling from Roon ARC, since that's all I use when listening on my iPhone (due to source enabling bugs in Roon Remote).

It appears that (1) Roon Arc listening doesn't show up at all in the sources, and (2) listening history is not exposed in the API. I'm going to open a thread on Roon Community and reach out to the rooExtend user and see if they have any ideas. But for me, not being able to include listend from Roon ARC makes this completely useless to me 😔

@MrPowerMac do you use Roon ARC? Are you still interested in a source that doesn't include those listens?

**Comment by MrPowerMac** [comment]
I would definitely like to have Roon ARC support, for the same reason as you. It’s my go to for any Roon listening on iOS even when I’m at home, and I also use it a lot in CarPlay since I’ve found it to be a nicer experience for Tidal/Qobuz than even their dedicated apps because of stuff like volume leveling.

**Comment by basepi** [comment]
OK, in that case, I'm going to shelve this for now. It lives in a [branch](https://github.com/basepi/multi-scrobbler/tree/codex/roon) on my fork, ready for me to pick back up if I can get the Roon team to give me API access to ARC listens.

I replied on the roon community thread, maybe I'll get a lead there.

**Comment by MrPowerMac** [comment]
A step forward for Roon scrobbling... the newest version 2.67 has added native ListenBrainz support!

...unfortunately it does not yet submit anything more than a single artist field, track name, and album name, just like a last.fm scrobble. So it's not much better as far as being able to tie into multi-scrobbler but I figured it'd at least be worth mentioning here. I've opened a feature request on the Roon forums for MBIDs to be submitted, so hopefully they consider it: https://community.roonlabs.com/t/submit-musicbrainz-ids-with-scrobbles-to-listenbrainz/320834

**Comment by FoxxMD** [comment]
That's great news! Does it allow specifying a custom LZ server? If so then it could fully integrate with multi-scrobbler via the [listenbrainz endpoint source](https://docs.multi-scrobbler.app/configuration/sources/listenbrainz-endpoint/)

**Comment by MrPowerMac** [comment]
It does not, unfortunately. The only input provided is adding your user token for submitting listens. I definitely would like to see custom server support eventually... might need to put in another feature request at some point.

**Comment by MrPowerMac** [comment]
I opened a feature request on the Roon forums for the ability to scrobble to custom ListenBrainz URLs: https://community.roonlabs.com/t/allow-scrobbling-to-custom-user-specified-listenbrainz-url/321060

Hopefully the Roon team sees it and decides it's a feature worth adding 🙂 Since Roon now has the ability to submit ListenBrainz-format scrobbles anyway I feel like it wouldn't be that hard for them to add...
